baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/OptionalOrElseMethodInvocationTests.java

```diff
@@ -0,0 +1,89 @@
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.Before;
+import org.junit.Test;
+
+public final class OptionalOrElseMethodInvocationTests {
+
+    private CompilationTestHelper compilationHelper;
+    private BugCheckerRefactoringTestHelper refactoringTestHelper;
+
+    @Before
+    public void before() {
+        compilationHelper = CompilationTestHelper.newInstance(OptionalOrElseMethodInvocation.class, getClass());
+        refactoringTestHelper = BugCheckerRefactoringTestHelper.newInstance(
+                new OptionalOrElseMethodInvocation(), getClass());
+    }
+
+    @Test
+    public void testNonCompileTimeConstantExpression() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import java.util.Optional;",
+                        "class Test {",
+                        "  String f() { return \"hello\"; }",
+                        "  // BUG: Diagnostic contains: non-constant expression",
+                        "  private final String string = Optional.of(\"hello\").orElse(f());",
+                        "}")
+                .doTest();
+    }
+
+    @Test
+    public void testNonCompileTimeConstantExpression_replacement() {
+        refactoringTestHelper
+                .addInputLines(
+                        "Test.java",
+                        "import java.util.Optional;",
+                        "class Test {",
+                        "  String f() { return \"hello\"; }",
+                        "  private final String string = Optional.of(\"hello\").orElse(f());",
+                        "}")
+                .addOutputLines(
+                        "Test.java",
+                        "import java.util.Optional;",
+                        "class Test {",
+                        "  String f() { return \"hello\"; }",
+                        "  private final String string = Optional.of(\"hello\").orElseGet(() -> f());",
+                        "}")
+                .doTest(BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH);
+    }
+
+    @Test
+    public void negative() {
+        compilationHelper
+                .addSourceLines(
+                        "Test.java",
+                        "import java.util.Optional;",
+                        "class Test {",
+                        "  private static final String compileTimeConstant = \"constant\";",
+                        "  String f() { return \"hello\"; }",
+                        "  void test() {",
+                        "    Optional.of(\"hello\").orElse(\"constant\");",
+                        "    Optional.of(\"hello\").orElse(compileTimeConstant);",
+                        "    String string = f();",
+                        "    Optional.of(\"hello\").orElse(string);",
```

Comment:
this is missing test coverage for `Optional.of(foo).orElseGet(() -> bar);`

Reply:
Not sure what you mean, you want this check to revert it to `orElse(bar)`?

Reply:
I don't see a test verifying that `Optional.of(foo).orElseGet(() -> bar)` is an acceptable thing to write

Reply:
sorry, I mean `Optional.of(foo).orElseGet(() -> f());`

Reply:
Ah, I've done that now